infra/deploy/build/Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM golang:1.18 as builder
+WORKDIR /home/platform-go-challenge
+COPY . ./
+
+RUN CGO_ENABLED=0 GOOS=linux go build -mod=vendor -a -installsuffix cgo -ldflags "-X main.version=$version" -o platform-go-challenge ./cmd/platform-go-challenge/main.go
```

Comment:
Why did you choose to use `-mod=vendor`? Also why do you have `CGO_ENABLED=0` and then `-installsuffix cgo` ?

Reply:
Hello @lunemec, thanks for the feedback!

When the -mod=vendor flag is used, go bulds the package using the vendor directory. There are 2 other options but they ignore the vendor directory. As I see prior to go 1.14 the -mod=vendor is the default value so it's not needed.

The reason CGO_ENABLED=0 is that we do not use any c code in our repo. CGO_ENABLED env variable is enabled by default and it can have an overhead in resources. 

As I searched, `installsuffix ` workaround was useful in go versions < 1.10 where it was an issue with static files that were not being regenerated.  Now the issue is fixed and that flag can be removed since we are on v1.18

more info here: (https://github.com/golang/go/issues/9344#issuecomment-69944514 )
and here: (https://twitter.com/bquenin/status/1131304803160604672) :laughing: